src/framework/filterControl/FilterItem.jsx

```diff
@@ -0,0 +1,122 @@
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import {
+  Filter,
+} from '../services';
+
+import Entity from '../services/string/Entity';
+import FilterForm from './forms/FilterForm.jsx';
+
+export default class FilterItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: this.props.filter,
```

Comment:
Don't know if keeping filter in state is needed as you get it passed anyway and when updated also pass it back to the parent (that then would update the prop I guess).